pkg/apis/externaldns/types.go

```diff
@@ -99,7 +99,7 @@ func (cfg *Config) ParseFlags(args []string) error {
 	// Flags related to providers
 	app.Flag("provider", "The DNS provider where the DNS records will be created (required, options: aws, google, azure, cloudflare, digitalocean, inmemory)").Required().PlaceHolder("provider").EnumVar(&cfg.Provider, "aws", "google", "azure", "cloudflare", "digitalocean", "inmemory")
 	app.Flag("google-project", "When using the Google provider, specify the Google project (required when --provider=google)").Default(defaultConfig.GoogleProject).StringVar(&cfg.GoogleProject)
-	app.Flag("domain-filter", "Limit possible target zones by a domain suffix (optional)").Default(defaultConfig.DomainFilter).StringVar(&cfg.DomainFilter)
+	app.Flag("domain-filter", "Limit possible target zones by a comma separeted list of domain suffixes (optional)").Default(defaultConfig.DomainFilter).StringVar(&cfg.DomainFilter)
```

Comment:
I would like to change this to use `StringVars`, note the traling `s`. This already supports processing multiple strings into a single string slice. This avoids `strings.Split(",")` in your code.

Furthermore, it changes the usage to provide the flag `--domain-filter` multiple times. Your example would become:

```
external-dns \
  --domain-filter=example.com \
  --domain-filter=company.org \
  --domain-filter=staging-company.org
```

This approach is also used by `helm` and `kubernetes`.